samples/99_zil_interpreter/app/builtins.rb

```diff
@@ -86,6 +123,45 @@ def define_for_evaled_arguments(&implementation)
   rand(range) + 1
 }
 
+ZIL_BUILTINS[:SET] = define_for_evaled_arguments { |arguments, context|
+  expect_argument_count!(arguments, 2)
+  var_atom = arguments[0]
+  context.locals[var_atom] = arguments[1]
+}
+
+ZIL_BUILTINS[:SETG] = define_for_evaled_arguments { |arguments, context|
+  expect_argument_count!(arguments, 2)
+  var_atom = arguments[0]
+  context.globals[var_atom] = arguments[1]
+}
+
+ZIL_BUILTINS[:BAND] = define_for_evaled_arguments { |arguments, _|
+  arguments.inject(&:&)
+  # while the book I'm reading tells me that these should take only two arguments
+  # I'll currently do it with `inject`, it's not a good idea but you were offline
+  # same with other `B...` methods
```

Comment:
Well let's follow ZIL specification and just give it 2 arguments... same with OR ;)

Reply:
yay, I can follow rules!